evalml/pipelines/utils.py

```diff
@@ -131,6 +131,33 @@ def _get_preprocessing_components(
     return pp_components
 
 
+def _make_component_dict_from_component_list(component_list):
+    """Generates a component dictionary from a list of components."""
+    components_with_names = []
+    seen = set()
+    for idx, component in enumerate(component_list):
+        component_name = component.name
+        if component_name in seen:
+            component_name = f"{component_name}_{idx}"
+        seen.add(component_name)
+        components_with_names.append((component_name, component))
+
+    component_dict = {}
+    most_recent_target = "y"
+    most_recent_features = "X"
+    for component_name, component_class in components_with_names:
```

Comment:
nit, but curious if it'd be possible to combine these for loops into 1 loop?

```python
seen = set()
component_dict = {}
most_recent_target = "y"
most_recent_features = "X"
for idx, component in enumerate(component_list):
    component_name = component.name
    if component_name in seen:
        component_name = f"{component_name}_{idx}"
    seen.add(component_name)

    component_dict[component_name] = [
        component,
        most_recent_features,
        most_recent_target,
    ]
    if component.modifies_target:
        most_recent_target = f"{component_name}.y"
    if component.modifies_features:
        most_recent_features = f"{component_name}.x"
return component_dict
```

Reply:
Yup, I like this! 😁